api/api.go

```diff
@@ -695,11 +705,31 @@ func matchFilterLabels(matchers []*labels.Matcher, sms map[string]string) bool {
 	for _, m := range matchers {
 		v, prs := sms[m.Name]
 		switch m.Type {
-		case labels.MatchNotEqual, labels.MatchNotRegexp:
+		case labels.MatchNotRegexp:
```

Comment:
I couldn't think of a "clever" way to do this, and didn't want to nest a bunch of if statements. Any suggestions are welcome, but this is pretty clear to read so I'm ok with it.